Velocity_Model/z.py

```diff
@@ -0,0 +1,178 @@
+"""
+Library functions for extracting Z values from the NZVM in a pythonic manner
+Supports NZVM versions 2.03, 2.04, 2.06
+"""
+from pathlib import Path
+import shutil
+import subprocess
+import tempfile
+
+import numpy as np
+import pandas as pd
+from matplotlib import path as mpltPath
+
+MAX_NUM_GEN_MULTI_PROFILES = 27000
+
+OUT_BASIN_SIGMA = 0.5
+
+IN_BASIN_SIGMA = 0.3
+
+file_output_format = {"Z1.0": "Z/Z_1.0.txt", "Z2.5": "Z/Z_2.5.txt"}
+column_names = {"Z1.0": "Z_1.0(km)", "Z2.5": "Z_2.5(km)"}
+
+
+def get_basin_full_path(basin_list):
+    vm_dir = Path(__file__).parent.parent.resolve()
+    return [vm_dir / basin for basin in basin_list]
+
+
+v203_basin_list = get_basin_full_path(
+    [
+        "Data/NI_BASINS/Wellington_Polygon_Wainuiomata_WGS84.txt",
+        "Data/Boundaries/NewCanterburyBasinBoundary_WGS84_1m.txt",
+        "Data/Boundaries/BPVBoundary.txt",
+        "Data/SI_BASINS/Cheviot_Polygon_WGS84.txt",
+        "Data/SI_BASINS/Hanmer_Polygon_WGS84.txt",
+        "Data/SI_BASINS/Kaikoura_Polygon_WGS84.txt",
+        "Data/SI_BASINS/Marlborough_Polygon_WGS84_v0p1.txt",
+        "Data/SI_BASINS/Nelson_Polygon_WGS84.txt",
+        "Data/SI_BASINS/NorthCanterbury_Polygon_WGS84.txt",
+        "Data/Boundaries/WaikatoHaurakiBasinEdge_WGS84.txt",
+    ]
+)
+v204_basin_list = v203_basin_list + get_basin_full_path(
+    [
+        "Data/USER20_BASINS/WanakaOutlineWGS84.txt",
+        "Data/USER20_BASINS/WakatipuBasinOutlineWGS84.txt",
+        "Data/USER20_BASINS/alexandra_outline.txt",
+        "Data/USER20_BASINS/ranfurly_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_A_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_B_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_C_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_D_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_E_outline.txt",
+        "Data/USER20_BASINS/bal_outline_WGS84.txt",
+        "Data/USER20_BASINS/dun_outline_WGS84.txt",
+        "Data/USER20_BASINS/mos_outline_WGS84.txt",
+        "Data/USER20_BASINS/Murchison_Basin_Outline_v1_WGS84.txt",
+        "Data/USER20_BASINS/hakataramea_outline_WGS84.txt",
+        "Data/USER20_BASINS/waitaki_outline_WGS84.txt",
```

Comment:
just a note, that these lines looks like the exact copy from the c codes.
And at some point in the future we might need to make the NZVM read a config file so other code/scripts can reuse it.

Reply:
Dibs not. :rofl: :sweat: 

But yeah, sensible suggestion.